utils/rgapi.d.ts

```diff
@@ -0,0 +1,266 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import type * as vscode from 'vscode';
+import type { AbstractAzExtTreeItem, AzExtParentTreeItem, AzExtTreeDataProvider, AzExtTreeItem, ISubscriptionContext, SealedAzExtTreeItem } from './index'; // This must remain `import type` or else a circular reference will result
+
+/**
+ * The API implemented by the Azure Resource Groups extension
+ */
+export interface AzureResourceGroupsExtensionApi {
+    /**
+     * The `AzExtTreeDataProvider` for the shared app resource view
+     * @deprecated Use `appResourceTree` instead
+     */
+    readonly tree: AzExtTreeDataProvider;
+
+    /**
+     * The `AzExtTreeDataProvider` for the shared app resource view
+     */
+    readonly appResourceTree: AzExtTreeDataProvider;

```

Comment:
This will still need to be implemented in the RG extension, but since there's two trees I thought it best we specify here _which_ tree.

Reply:
Just to make sure I understand, the other tree would be the workspace tree?

Reply:
Yeah. Should I rename it "workspaceTree" instead of "localResourceTree"?

Reply:
Yeah go with "workspaceTree" for now, we may end up changing the wording later since workspace is already a separate thing in VS Code-land. Could be confusing for users.

Reply:
I did that; `local` is replaced with `workspace`. I also sectioned out the deprecated parts so it's trivial to remove them later.